superset/assets/src/visualizations/deckgl/CategoricalDeckGLContainer.jsx

```diff
@@ -46,57 +46,64 @@ export default class CategoricalDeckGLContainer extends React.PureComponent {
    * The container will have an interactive legend, populated from the
    * categories present in the data.
    */
+  constructor(props) {
+    super(props);
 
-  /* eslint-disable-next-line react/sort-comp */
-  static getDerivedStateFromProps(nextProps) {
-    const fd = nextProps.slice.formData;
-
+    const fd = props.payload.form_data;
```

Comment:
Why use `payload.form_data` instead of `slice.formData` in the previous code?
Also, line 77, use `slice.formData`. Shouldn't both be using the same `formData`?

Reply:
The difference is that `payload.form_data` is updated by the filter box, but `slice.formData` is not. 

I changed this because I found a bug where the filter box was not sending a new time granularity to a chart. I had a chart with a saved granularity of 1 day, and I changed the filter box granularity to be 1 hour. If we look at the two form data objects here when the filter box changes the granularity, their value would be:

```javascript
console.log(props.payload.form_data.time_grain_sqla);  // PT1H
console.log(props.slice.formData.time_grain_sqla);  // P1D
```

In the explore view, the values are always the same.

I think we should always use `payload.form_data` instead. Line 77 probably works because it uses values that are not being changed by a filter box. I'll make the change there and test the code.

Reply:
I also don't have the full story here, maybe @mistercrunch can chime in why the duplication?

You're right, we should investigate why `slice.formData` is not being updated (assuming it should). I'll take a look.

Reply:
@betodealmeida @kristw wanted to make sure you saw ^